### PR TITLE
LL-6112 (DS Setup) LL-6344 LL-6345 LL-6346 LL-6347 LL-6348 LL-6499

### PR DESCRIPTION
**Pull request by LFBarreto**
(DS Setup): Added uilib + storybook + Text base component
### Type

Feature
### Context

Main issue: [LL-6112]

[LL-6344]
[LL-6345]
[LL-6346]
[LL-6347]
[LL-6348]
[LL-6499]
[LL-6497]
[LL-6498]
### Parts of the app affected / Test plan

Should not affect the app for now since its just adding the building blocks for the next DS integrations
run it:
```
cd uilib 
rm -rf node_modules
yarn
yarn web
```
on another terminal
```
cd uilib
yarn storybook
```

Install [Expo Go app](https://expo.dev/client) on your phone and scan the QR code to run it on a real device or run it on an emulator using 
```
cd uilib
yarn android (or yarn ios)
```

[LL-6112]: https://ledgerhq.atlassian.net/browse/LL-6112

[LL-6344]: https://ledgerhq.atlassian.net/browse/LL-6344
[LL-6345]: https://ledgerhq.atlassian.net/browse/LL-6345
[LL-6346]: https://ledgerhq.atlassian.net/browse/LL-6346
[LL-6347]: https://ledgerhq.atlassian.net/browse/LL-6347
[LL-6499]: https://ledgerhq.atlassian.net/browse/LL-6499

[LL-6348]: https://ledgerhq.atlassian.net/browse/LL-6348
[LL-6497]: https://ledgerhq.atlassian.net/browse/LL-6497
[LL-6498]: https://ledgerhq.atlassian.net/browse/LL-6498

### CODE REVIEW

File: uilib/lib/icons/BracketLeft.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import React from 'react';
+declare const BracketLeft: (props: any) => React.ReactElement;
+export default BracketLeft;
```
Comment (jgiamporcaro-ledger) [inline uilib/lib/icons/BracketLeft.d.ts:3]: If I'm right, `react-native-svg` library exposes props type for his SVG component.

```suggestion
import React from 'react';
import type {  SvgProps  } from 'react-native-svg';

declare const BracketLeft: (props: SvgProps) => React.ReactElement;
export default BracketLeft;
```

**Review by juan-cortes — CHANGES_REQUESTED** [review]
Seems like we are committing both /src and /lib, shouldn't lib be ignored?

**Comment by gre** [comment]
> these are the rules used in the TS migration on live-common 👍

that said, we're not yet migrating TS inside LLM, are we? i'm lost a bit why you took TS live-common work in this as that TS live-common work is not even merged in & TS itself is not planned to rework on LLM yet (possibly after your yellow impacting work is finished indeed)

**Comment by LFBarreto** [comment]
> > these are the rules used in the TS migration on live-common +1
> 
> that said, we're not yet migrating TS inside LLM, are we? i'm lost a bit why you took TS live-common work in this as that TS live-common work is not even merged in & TS itself is not planned to rework on LLM yet (possibly after your yellow impacting work is finished indeed)
@gre 
We are developing the UI lib in TS to not have to port it afterwards since its a separate project in itself why not start directly in TS :shrug: 
So i reused the TS lint config we're using on live common for now and we might need to adapt as we go.
the lib will be built using tsc and we will target the compiled JS files on the main app once its ready to use :+1:

**Comment by ggilchrist-ledger** [comment]
@LFBarreto  @juan-cortes @jelbaz-ledger @jgiamporcaro-ledger I can’t get the mobile app to run normally with yarn android in the root folder for Android (physical/emulator). iOS is fine. Not sure the error message is too useful but it is:
```
> Task :ledgerhq_react-native-hid:compileDebugJavaWithJavac FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.2/userguide/command_line_interface.html#sec:command_line_warnings
12 actionable tasks: 2 executed, 10 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':ledgerhq_react-native-hid:compileDebugJavaWithJavac'.
> Could not find tools.jar. Please check that /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jre/Contents/Home contains a valid JDK installation.
```

File: uilib/src/components/Text/getTextStyle.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  lineHeight: number;
+  fontWeight?: number;
+  textDecoration?: string;
+  paddingTop?: number;
+} {
+  switch (type) {
+    case "h1":
+      return {
+        fontFamily: "Alpha",
+        fontSize: 36,
```
Comment (ggilchrist-ledger) [inline uilib/src/components/Text/getTextStyle.ts:33]: How are we getting the font sizes and line heights for these? They don't match with the Figma's @LFBarreto @juan-cortes @jgiamporcaro-ledger

File: uilib/src/components/Text/getTextStyle.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        lineHeight: 43.2,
+        paddingTop: bracket ? 15 : 0,
```
Comment (ggilchrist-ledger) [inline uilib/src/components/Text/getTextStyle.ts:35]: I think this is causing the following:

**Comment by ggilchrist-ledger** [comment]
I've added [this](https://ledgerhq.atlassian.net/secure/RapidBoard.jspa?rapidView=277&modal=detail&selectedIssue=LL-6497&sprint=719) since the Flex story is part of this PR

**Comment by ggilchrist-ledger** [comment]
Added [this](https://ledgerhq.atlassian.net/secure/RapidBoard.jspa?rapidView=277&projectKey=LL&modal=detail&selectedIssue=LL-6498&sprint=719) too since it's included

**Pull request merged**